File: frida-hello/hook.js
'use strict';
const frida = require('frida');

function agent() {
  // const pointer = ptr('0x106863ed0');
  const pointer = Module.findExportByName(null, 'f');
  Interceptor.attach(pointer, {
    onEnter: function (args) {
      args[0] = ptr('1337');
    }
  });

  rpc.exports = {
    add: function (a, b) {
      return a + b;
    },
    sub: function (a, b) {
      return new Promise(function (resolve) {
        setTimeout(function () {
          resolve(a - b);
        }, 100);
      });
    }
  };

  send(123);

  recv('poke', function onMessage(pokeMessage) {
    send('pokeBack');
  });
}

async function main() {
  const session = await frida.attach('hello');
  const script = await session.createScript('(' + agent.toString() + ').call(this);');
  script.events.listen('message', (message, data) => {
    console.log(message);
  });

  await script.load();

  await script.post({ type: 'poke' });

  const api = await script.getExports();
  console.log(await api.add(2, 3));
  console.log(await api.sub(5, 3));
}

main().catch(console.error);

File: frida-hello/hook.py
from __future__ import print_function
import frida
import sys

session = frida.attach("hello")

hook_script = session.create_script("""
Interceptor.attach(ptr("%s"), {
    onEnter: function(args) {
        send(args[0].toInt32());
    }
});
""" % int(sys.argv[1], 16))
def on_message(message, data):
    print(message)
hook_script.on('message', on_message)
hook_script.load()


modify_script = session.create_script("""
Interceptor.attach(ptr("%s"), {
    onEnter: function(args) {
        args[0] = ptr("1337");
    }
});
""" % int(sys.argv[1], 16))
# modify_script.load()


call_script = session.create_script("""
var f = new NativeFunction(ptr("%s"), 'void', ['int']);
f(1911);
f(1911);
f(1911);
""" % int(sys.argv[1], 16))
call_script.load()


sys.stdin.read()
